Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -412,98 +414,104 @@ void SourceBufferPrivate::evictCodedFrames(uint64_t newDataSize, uint64_t maximu
     // This algorithm is run to free up space in this source buffer when new data is appended.
     // 1. Let new data equal the data that is about to be appended to this SourceBuffer.
     // 2. If the buffer full flag equals false, then abort these steps.
-    if (!isBufferFullFor(newDataSize, maximumBufferSize))
+    bool isBufferFull = isBufferFullFor(newDataSize, maximumBufferSize);
+    if (!isBufferFull)
         return;
 
     // 3. Let removal ranges equal a list of presentation time ranges that can be evicted from
     // the presentation to make room for the new data.
 
-    // NOTE: begin by removing data from the beginning of the buffered ranges, 30 seconds at
-    // a time, up to 30 seconds before currentTime.
-    MediaTime thirtySeconds = MediaTime(30, 1);
-    MediaTime maximumRangeEnd = currentTime - thirtySeconds;
+    // NOTE: begin by removing data from the beginning of the buffered ranges, timeChunk seconds at
+    // a time, up to timeChunk seconds before currentTime.
 
 #if !RELEASE_LOG_DISABLED
     uint64_t initialBufferedSize = totalTrackBufferSizeInBytes();
     DEBUG_LOG(LOGIDENTIFIER, "currentTime = ", currentTime, ", require ", initialBufferedSize + newDataSize, " bytes, maximum buffer size is ", maximumBufferSize);
 #endif
 
-    MediaTime rangeStart = MediaTime::invalidTime();
+    const auto& buffered = m_buffered->ranges();
 
-    for (auto& trackBuffer : m_trackBufferMap.values()) {
-        auto iter = trackBuffer->samples().presentationOrder().findSampleContainingOrAfterPresentationTime(MediaTime::zeroTime());
-        if (iter != trackBuffer->samples().presentationOrder().end()) {
-            MediaTime startTime = iter->first;
-            if (rangeStart.isInvalid() || startTime < rangeStart)
-                rangeStart = startTime;
-        }
-    }
-
-    if (rangeStart.isInvalid())
-        rangeStart = MediaTime::zeroTime();
+    unsigned timeChunkAsMilliseconds = evictionAlgorithmInitialTimeChunk;
+    do {
+        MediaTime timeChunk = MediaTime(timeChunkAsMilliseconds, 1000);
+        MediaTime rangeStart = buffered.minimumBufferedTime();
+        MediaTime rangeEnd = rangeStart + timeChunk;
+        MediaTime maximumRangeEnd = currentTime - timeChunk;
+        while (rangeStart < maximumRangeEnd) {
+            // 4. For each range in removal ranges, run the coded frame removal algorithm with start and
+            // end equal to the removal range start and end timestamp respectively.
+            MediaTime realRangeEnd = std::min(rangeEnd, maximumRangeEnd);
+            if (rangeStart >= realRangeEnd)
+                break;
```

Comment:
This is the first and last read of `rangeEnd`. Therefore, we have no need for a `realRangeEnd` variable. We could instead inline the `min()` expression in the declaration of `rangeEnd`, also removing the need for the extra loop iteration guard here.

Reply:
I'll check this

Reply:
You need to inline this twice because you'd have to write it twice. IMHO, it is more readable like this.

Anyway, there are some more `const`s I can add.

Reply:
Where would you have to write it twice?

> IMHO, it is more readable like this.
I agree doing things in two lines is more readable, but having a variable named *thing* and another called *realThing* living in the same scope is a footgun.

Reply:
Maybe rename it to `clampedThing`

Reply:
That is indeed better wording. Specific, conveys meaning. Still, having two variables that can be easily confused for each other, that often but not always hold the same value, for too long in scope, is still a footgun.

Reply:
I prefer not to change it further. The name should be enough, IMHO.